### PR TITLE
PN-82: reveal disorder for matchable cases

### PR DESCRIPTION
**Pull request by buske**
- RestrictedDisorderSimilarityView now extends DefaultDisorderSimilarityView

### CODE REVIEW

File: similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/RestrictedDisorderSimilarityView.java [hunk 1]
```diff
@@ -52,73 +45,31 @@
...
-        this.reference = reference;
+        super(match, reference);
         this.access = access;
     }
 
     @Override
     public String getId()
     {
-        return this.access.isOpenAccess() && this.match != null ? this.match.getId() : null;
+        return this.access.isPrivateAccess() ? null : super.getId();
```
Comment (sdumitriu) [inline similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/RestrictedDisorderSimilarityView.java]: Do we want to disclose the diagnosis, or say "Undisclosed"?

**Comment by sdumitriu** [comment]
Other than the question of disclosed vs undisclosed, this looks good.

File: similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/RestrictedDisorderSimilarityView.java (hunk 1 above)
Comment (buske) [inline similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/RestrictedDisorderSimilarityView.java]: Yep, I think we want to go ahead and disclose the diagnosis for matchable cases.

**Comment by buske** [comment]
Thanks for taking a look, @sdumitriu.

**Pull request merged**